src/Bicep.Cli/CommandLine/Arguments/ArgumentsBase.cs

```diff
@@ -4,5 +4,6 @@ namespace Bicep.Cli.CommandLine.Arguments
 {
     public abstract class ArgumentsBase
     {
+        public string? CommandName;
```

Comment:
>CommandName [](http://example.com/codeflow?start=23&length=11)

Can you also make this immutable? (similar properties in the derived classes only have a getter and are initialized in the constructor)

Reply:
Yes, sounds better too. I will update as soon as possible.

Reply:
Hello @majastrz. One doubt, you mean somethink like this?

```csharp
public abstract class ArgumentsBase
{
    public string? CommandName { get; }

    protected ArgumentsBase(string commandName)
    {
        CommandName = commandName;
    }
}
```

And then the classes like:
```csharp
public BuildOrDecompileArguments(string[] args, string commandName) : base(commandName)
```
and for the others:
```csharp
public HelpArguments(string commandName) : base(commandName)
```

Reply:
Yup, exactly.

Reply:
Thank you @majastrz ! Just uploaded this last change 😃

Reply:
Awesome!